bindings/matrix-sdk-ffi/src/platform.rs

```diff
@@ -145,15 +171,34 @@ where
     Layer::and_then(
         file_layer,
         config.write_to_stdout_or_system.then(|| {
+            // Another fields formatter is necessary because of this bug
+            // https://github.com/tokio-rs/tracing/issues/1372. Using a new
+            // formatter for the fields forces to record them in different span
+            // extensions, and thus remove the duplicated fields in the span.
+            #[derive(Default)]
+            struct FieldsFormatterFormStdoutOrSystem(DefaultFields);
+
+            impl<'writer> FormatFields<'writer> for FieldsFormatterFormStdoutOrSystem {
+                fn format_fields<R: RecordFields>(
+                    &self,
+                    writer: Writer<'writer>,
+                    fields: R,
+                ) -> std::fmt::Result {
+                    self.0.format_fields(writer, fields)
+                }
+            }
```

Comment:
nit: we don't need to duplicate the struct definition, if we hoist it outside the functions and reuse it in both places. A better name would need to be found then.

Reply:
indeed!

Reply:
As tested by @stefanceriu in https://github.com/matrix-org/matrix-rust-sdk/pull/3851#issuecomment-2296453939, using the same type sadly doesn't work. Let's use 2 types then.